agent/logging.go

```diff
@@ -167,3 +171,36 @@ func (a *Agent) prepSingleLogger(tc *taskContext, in model.LogOpts, logDir, file
 		Filepath:        filepath.Join(logDir, fileName),
 	}
 }
+
+func redactList(projectVars map[string]string, redacted map[string]bool) []string {
+	suspiciousPatterns := []string{
+		"key",
+		"secret",
+		"auth",
+		"token",
+		"private",
+		"pass",
+		"pw",
+	}
+
+	var redactedKeys []string
+	for key := range projectVars {
+		if _, ok := redacted[key]; ok {
+			// Skip since the key is already in the redacted list.
+			continue
+		}
+
+		for _, pattern := range suspiciousPatterns {
+			if strings.Contains(strings.ToLower(key), pattern) {
```

Comment:
Why don't we also check the values for suspicious patterns? For example, we can look for things that are often at the start of certain keys like `ghs_` for github.

Reply:
To get this out and deployed I think we should just start with this set of patterns and expand as we see fit.

Reply:
My suggestion is not just to expand the set of patterns, but to scan both keys and values. If getting this out quickly is the priority, would it maybe make sense to do this in a separate ticket?

Reply:
Yep, that can be done here as well. We can add more scanning but the hope is that users will also rotate their secrets and mark their sensitive vars private.